File: Functions/update_budget.py
print("This is a budget calculator")

def inputs(ty):
   
  
   variable = (int(input(f"What is your {ty} by month?")))
   return variable

 
income = inputs("income")
rent = inputs("rent")
utilities = inputs("utilities")
grocieries = inputs("grocieries")
car = inputs("transportation")
   


expenses = rent + utilities + grocieries + car
savings = income
spend = income - expenses
print("Your monthly income is", income)
print("Your expenses are", expenses)
print("Your monthly savings is", savings)
print("Your spending money is", spend)
prent = (rent/income) * 100
pgroceries = (grocieries/ income)* 100
psaving = (savings/ income) * 100
pexpenses = (expenses/ income) * 100
putilities= (utilities/ income) * 100
pcar = (car/income) * 100

def percent(type, amount):
    per = amount/income *100
    print(f"Your {type} is {per}% of your income.")

percent("rent", rent)
percent("utilities", utilities)
percent("grocieries", grocieries)
percent("transportation", car)
percent("expenses", expenses)
percent("savings", savings)

File: Functions/update_budget.c
#include <stdio.h>
float income, rent, utilities, grocieries, car, savings, expenses, spend;

void percent(char type[], float amount){
  float per = amount/income *100;

  printf("Your %s is %f of your income. \n", type, per);
}

float inputs(char type[],float var){
  printf("monthly %s:\n", type);
  scanf("%f", &var);
  return var;

}

int main(void){ 
    
 printf("This is a budget calculator.\n");
 income = inputs("income", income);
 rent = inputs("rent", rent);
 utilities = inputs("utilities", utilities);
 grocieries = inputs("grocieries", grocieries);
 car = inputs("transportation", car);

 savings = income *.2;
 expenses = rent + utilities + grocieries + car;
 spend = income - savings - expenses ; 
 printf("you make %.2f\n",income);
 printf("Your expenses are %.2f \n", expenses);
 printf ("Your savings are %.2f \n", savings);
 printf("Your spending money is %.2f \n", spend);
   
 percent("rent ", rent);
 percent("utilities",utilities);
 percent("grocieries", grocieries);
 percent("transportation", car);
 percent("savings", savings);
 percent("expences", expenses);
 percent("spend money", spend);
    
return 0; 

}